### PR TITLE
Client gen

### PR DESCRIPTION
**Pull request by davegolland**

### CODE REVIEW

File: gen-clients.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
```
Comment (w01fe) [inline gen-clients.sh]: :sheep: maybe put in `bin/`rather than at toplevel?

**Comment by w01fe** [comment]
Awesome!  The zip is nice for downloading, but also makes browsing the source of the clients potentially more difficult -- not sure what's conventional?  

:shipit:

**Comment by davegolland** [comment]
Hm, we could do both, source and zip. It just felt weird adding so many
files. Also, not sure whether adding the files might open us up to PRs
modifying the generated files.

I agree that for browsing the unzipped files might be good. I'll do both.

**Comment by w01fe** [comment]
Makes sense -- this seems good for now then.

**Pull request merged**